### PR TITLE
Long needed maintenance

### PR DESCRIPTION
**Pull request by chdanielmueller**
Based upon #37 and #45.
Big Thanks to @shahyar and @ddimitrioglo which already did most of the work.

I will now add my spin on it and some changes to processes.

### CODE REVIEW

**Comment by ddimitrioglo** [comment]
@chdanielmueller Thanks for allocating your time on this. Next week we are going to test our feature and we will try to use v2.0. Will give you the feedback in a week or so!

File: jest.config.js [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
+  clearMocks: true,
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+  // The directory where Jest should output its coverage files
+  coverageDirectory: 'coverage',
+  // Indicates which provider should be used to instrument code for coverage
   coverageProvider: 'v8',
-  moduleFileExtensions: ['js', 'json', 'ts'],
-  testRegex: ['.test.ts$'],
-  transform: {
```
Comment (ddimitrioglo) [inline jest.config.js]: if you have switched to `ts-jest` you can remove `swc` from the dependencies

File: package.json [hunk 2]
```diff
@@ -58,6 +56,7 @@
     "@typescript-eslint/eslint-plugin": "6.19.0",
     "@typescript-eslint/parser": "6.19.0",
     "amqplib": "0.10.3",
+    "del-cli": "5.1.0",
```
Comment (ddimitrioglo) [inline package.json:57]: just FYI: del-cli > del > rimraf

File: jest.config.js (hunk 1 above)
Comment (chdanielmueller) [reply jest.config.js]: Done that. Thanks

File: package.json (hunk 2 above)
Comment (chdanielmueller) [reply package.json:57]: Didn't know that. Thanks

**Comment by chdanielmueller** [comment]
I was able to test it successfully in my own project and in #44 it has been confirmed that it works as well.
I would call that a success and merge this PR and promote the version to latest.

**Comment by ddimitrioglo** [comment]
@chdanielmueller Hey, we are still in a testing process but so far - looks good 👍

**Pull request merged**